Date: 2005-08-27
Initial Package Version: 2.9.6
Upstream Status: Frugalware-specific, not submitted.
Origin: Christian Hamar alias krix <krics@linuxforum.hu> and
        Miklos Vajna <vmiklos@frugalware.org>
Description: Fixes three annoying bugs in pacman's option parsing.
 1) If the operation requires at least one target, display the operation's help.
    (eg. pacman -R is not a valid command)
 2) If there is an unknown option, after printfing the error message display
    the operation's help.
 3) fprintf an error message to stderr + display help if no operation specified,
    but would have to. (eg. -h or -V not used)

Examples:
Invalid options        Original            Patched
 -Qx                   invalid option      invalid option + Q's help
 -A -R                 only one operation  unchanged
 -R                    silent              -R's help
 foo                   slient              general help

This diff has been accepted by upstream except the chroot part as upstream does
not use chroot. Maybe later could be merged with -chroot.diff
diff -Naur pacman-2.9.6.orig/scripts/makepkg pacman-2.9.6/scripts/makepkg
--- pacman-2.9.6.orig/scripts/makepkg	2005-08-28 19:28:42.000000000 +0200
+++ pacman-2.9.6/scripts/makepkg	2005-08-28 19:33:14.000000000 +0200
@@ -286,10 +286,12 @@
 				removelist="$removelist $i"
 			fi
 		done
-		pacman -r $CHROOTDIR -Rc $removelist --noconfirm
-		if [ "$?" != "0" ]; then
-			error "Failed to remove packages."
-			exit 1
+		if [ ! -z "$removelist" ]; then
+			pacman -r $CHROOTDIR -Rc $removelist --noconfirm
+			if [ "$?" != "0" ]; then
+				error "Failed to remove packages."
+				exit 1
+			fi
 		fi
 	fi
 }
